<launch>
    <!-- Handle detection configuration -->
    <arg name="weights" default="$(find qm_vision)/models/yolov8_handle.pt"/>
    <arg name="confidence_threshold" default="0.75"/>
    <arg name="iou_threshold" default="0.45"/>
    <arg name="maximum_detections" default="1"/>
    <arg name="threshold_l"      default="30"/>
    <arg name="threshold_u"      default="255"/>

    <!-- Door detection configuration -->
    <arg name="dweights" default="$(find qm_vision)/models/yolov8_door.pt"/>
    <arg name="dconfidence_threshold" default="0.75"/>
    <arg name="diou_threshold" default="0.45"/>
    <arg name="dmaximum_detections" default="1"/>
    <arg name="dthreshold_l"      default="30"/>
    <arg name="dthreshold_u"      default="255"/>

    <!-- Visualize using OpenCV window -->
    <arg name="view_image" default="true"/>
    <!-- Publish images to ros topic -->
    <arg name="publish_image" default="true"/>
    
    <!-- Input data topics -->
    <arg name="input_image_topic" default="/camera/color/image_raw"/>
    <arg name="camera_info_topic" default="/camera/depth/camera_info"/> 
    <arg name="input_depth_topic" default="/camera/depth/image_rect_raw"/>
    <arg name="input_pc_topic" default="/camera/depth/color/points"/>

    <!-- Module topics-->
    <arg name="output_image_topic" default="/yolo/image"/>
    <arg name="output_mask_topic" default="/yolo/mask"/>
    <arg name="output_depth_topic" default="/yolo/depth_registered"/>
    <arg name="output_masked_depth_nr_topic" default="/yolo/depth_masked"/>
    <arg name="output_masked_depth_topic" default="/yolo/masked_depth_registered"/>  
    <arg name="output_pc_topic" default="/handle/points"/>
    <arg name="output_centroid_topic" default="/handle/centroid"/>
    <arg name="output_orientation_topic" default="/handle/orientation"/>
    <arg name="output_width_topic" default="/door/pixel_width"/>

    <node pkg="qm_vision" name="plane_extractor_node" type="plane_extractor.py" output="screen">
        <param name="input_pc_topic" value="$(arg input_pc_topic)"/>
    </node>



</launch>